code/game/atoms.dm

```diff
@@ -1392,6 +1394,32 @@
 
 		SEND_SIGNAL(src, COMSIG_ATOM_VV_MODIFY_TRANSFORM)
 
+	if(href_list[VV_HK_SPIN_ANIMATION] && check_rights(R_VAREDIT))
+		var/num_spins = input(usr, "Do you want infinite spins?", "Spin Animation") in list("Yes", "No")
+		if(num_spins == "No")
+			num_spins = input(usr, "How many spins?", "Spin Animation") as null|num
+		else
+			num_spins = -1
+		if(!num_spins)
+			return
+		var/spin_speed = input(usr, "How fast?", "Spin Animation") as null|num
+		if(!spin_speed)
+			return
+		var/direction = input(usr, "Which direction?", "Spin Animation") in list("Clockwise", "Counter-clockwise")
+		if(!direction)
+			return
+		if(direction == "Clockwise")
+			direction = 1
+		else
+			direction = 0
```

Comment:
```suggestion
		switch(direction)
			if("Clockwise")
				direction = 1
			if("Counter-clockwise")
				direction = 0
			else
				return
```

Let's do it this way such that we don't explicitly look for a falsey value, but such that we assume that any non-wanted value indicates the user wanting to exit the proc. I doubt that your current implementation would break or be weird, but lets be safe

Reply:
Agree, didn't think of this either